clients/unshaded/src/main/java/tachyon/client/file/FileSystem.java

```diff
@@ -0,0 +1,375 @@
+      throws FileAlreadyExistsException, InvalidPathException, IOException, TachyonException;
+
+  /**
+   * Convenience method for {@link #delete(TachyonURI, DeleteOptions)} with default options.
+   *
+   * @param path the path to delete in Tachyon space
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given path does not exist
+   * @throws DirectoryNotEmptyException if recursive is false and the path is a nonempty directory
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  void delete(TachyonURI path)
+      throws DirectoryNotEmptyException, FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Deletes a file or a directory.
+   *
+   * @param path the path to delete in Tachyon space
+   * @param options options to associate with this operation
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given path does not exist
+   * @throws DirectoryNotEmptyException if recursive is false and the path is a nonempty directory
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  void delete(TachyonURI path, DeleteOptions options)
+      throws DirectoryNotEmptyException, FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Convenience method for {@link #exists(TachyonURI, ExistsOptions)} with default options.
+   *
+   * @param path the path in question
+   * @return true if the path exists, false otherwise
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws InvalidPathException if the path is invalid
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  boolean exists(TachyonURI path) throws InvalidPathException, IOException, TachyonException;
+
+  /**
+   * Checks whether a path exists in Tachyon space
+   *
+   * @param path the path in question
+   * @param options options to associate with this operation
+   * @return true if the path exists, false otherwise
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws InvalidPathException if the path is invalid
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  boolean exists(TachyonURI path, ExistsOptions options)
+      throws InvalidPathException, IOException, TachyonException;
+
+  /**
+   * Convenience method for {@link #free(TachyonURI, FreeOptions)} with default options.
+   *
+   * @param path the path to free in Tachyon space
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given path does not exist
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  void free(TachyonURI path) throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Evicts any data under the given path from Tachyon space, but does not delete the data from the
```

Comment:
Updated. It is ok for the path to be a directory. I try to use path when specifying file/directory, but it is hard to pick up on since the style is not yet consistent.